### PR TITLE
Add tests to `test_datastructure`

### PR DESCRIPTION
**Pull request by TechNiick**
# Summary

Add test cases to `test_datastructures.py` in order to have 100 coverage. Related to [this issue](https://github.com/encode/starlette/issues/2452) and [this PR](https://github.com/encode/starlette/pull/2503).

# Checklist

- [ ] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [ ] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [ ] I've updated the documentation accordingly.

### CODE REVIEW

File: tests/test_datastructures.py [hunk 1]
```diff
@@ -58,6 +58,9 @@ def test_url() -> None:
     url = URL("http://u:p@host:80")
     assert url.replace(port=88) == URL("http://u:p@host:88")
 
+    url = URL("http://host:80")
+    assert url.replace(username="user").password is None
```
Comment (Kludex) [inline tests/test_datastructures.py]: What are you testing here?

File: tests/test_datastructures.py [hunk 2]
```diff
@@ -70,6 +73,10 @@ def test_url_query_params() -> None:
     assert str(u) == "https://example.org/path/?order=name"
     u = u.remove_query_params("order")
     assert str(u) == "https://example.org/path/"
+    u = u.include_query_params(page=4, search="testing")
+    assert str(u) == "https://example.org/path/?page=4&search=testing"
+    u = u.remove_query_params(["page", "search"])
+    assert str(u) == "https://example.org/path/"
```
Comment (Kludex) [inline tests/test_datastructures.py:79]: What are you testing here?

Better to post the lines you are testing, otherwise, this will take some time for me.

**Review by Kludex — CHANGES_REQUESTED** [review]

**Comment by TechNiick** [comment]
This test: 

https://github.com/encode/starlette/blob/4dcdbc4a347a0613a92f5030273eab64381769b5/tests/test_datastructures.py#L61-L62 
Is used for testing the following lines:

https://github.com/encode/starlette/blob/74ccb961b3c1b1871aa7ed70a81dd3000e0194da/starlette/datastructures.py#L134-L136

**Comment by TechNiick** [comment]
This test: 

https://github.com/encode/starlette/blob/4dcdbc4a347a0613a92f5030273eab64381769b5/tests/test_datastructures.py#L76-L79
Is used for testing the following lines:

https://github.com/encode/starlette/blob/74ccb961b3c1b1871aa7ed70a81dd3000e0194da/starlette/datastructures.py#L154-L156

**Comment by TechNiick** [comment]
This test: 

https://github.com/encode/starlette/blob/4dcdbc4a347a0613a92f5030273eab64381769b5/tests/test_datastructures.py#L148-L160
Is used for testing the following lines:

https://github.com/encode/starlette/blob/74ccb961b3c1b1871aa7ed70a81dd3000e0194da/starlette/datastructures.py#L39-L40

File: tests/test_datastructures.py [hunk 3]
```diff
@@ -138,6 +145,19 @@ def test_url_from_scope() -> None:
...
+        "query_string": b"query=string",
+        "headers": [
+            (b"content-type", b"text/html"),
+            (b"host", b"example.com:8000"),
+            (b"accept", b"text/html"),
+        ],
+    }
+    url_from_scope_with_host = URL(scope=scope_with_host_header)
+    assert url_from_scope_with_host.hostname == "example.com"
+
```
Comment (Kludex) [inline tests/test_datastructures.py]: I meant doing it here... 
https://github.com/encode/starlette/blob/74ccb961b3c1b1871aa7ed70a81dd3000e0194da/starlette/datastructures.py#L39-L40

**Review by Kludex — APPROVED** [review]
Thanks! 👍 

I've updated the test cases to match the current pattern we have there.

**Pull request merged**